soh/src/code/z_actor.c

```diff
@@ -3464,6 +3466,8 @@ Actor* Actor_Delete(ActorContext* actorCtx, Actor* actor, PlayState* play) {
     Actor* newHead;
     ActorDBEntry* dbEntry;
 
+    GameInteractor_ExecuteOnActorDelete(actor);
```

Comment:
I was unsure where to put this in the function, but I chose right at the start because, since this function destroys and eventually frees the actor, it make sense for hooks to respond to it with as much information about it as possible.